lib/SubjectByPropertyValueHandler.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const { namespaceMap } = require( './rdfNamespaces' );
+const CirrusSearchRequest = require( './CirrusSearchRequest' );
+
+module.exports = class SubjectByPropertyValueHandler {
+
+	handle( _query, parsedQuery ) {
+		if ( !parsedQuery ||
+			parsedQuery.queryType !== 'SELECT' ||
+			!parsedQuery.where ||
+			parsedQuery.where.length !== 1 ||
+			parsedQuery.where[ 0 ].triples.length !== 1 ) {
+			return false;
+		}
+
+		const triple = parsedQuery.where[ 0 ].triples[ 0 ];
+		if ( triple.subject.termType !== 'Variable' ||
+			triple.predicate.termType !== 'NamedNode' ||
+			!triple.predicate.id.startsWith( namespaceMap.Wikidata.wdt ) ) {
+			return false;
+		}
+
+		let value;
+		if ( triple.object.termType === 'Literal' && /^"\w+"$/.test( triple.object.id ) ) {
+			value = triple.object.id.substring( 1, triple.object.id.length - 1 );
+		} else if ( triple.object.termType === 'NamedNode' &&
+			triple.object.id.startsWith( namespaceMap.Wikidata.wd ) ) {
+			value = triple.object.id.substring( namespaceMap.Wikidata.wd.length );
+		} else {
+			return false;
+		}
+
+		return new CirrusSearchRequest(
+			triple.subject.id.substring( 1 ),
```

Comment:
Maybe there should also be a check that the `SELECT` has exactly this variable and nothing else? So that we don’t match (hypothetical) queries like `SELECT ?item (NOW() AS ?asOf) WHERE { ?item wdt:P123 "456" }`

Reply:
Yup good point! Done!